src/amo/components/AddonReviewCard/index.js

```diff
@@ -371,38 +378,66 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
         );
       }
 
-      const byLineLink = replaceStringsWithJSX({
-        text: i18n.sprintf(byLineString, {
-          authorName: review.userName,
-          timestamp: i18n.moment(review.created).fromNow(),
-          // Keep the link placeholders so that we can use them to inject a
-          // `<Link />` using `replaceStringsWithJSX`.
-          linkEnd: '%(linkEnd)s',
-          linkStart: '%(linkStart)s',
-        }),
-        replacements: [
-          [
-            'linkStart',
-            'linkEnd',
+      const arrReplacements = [
+        [
+          'linkStart',
+          'linkEnd',
+          (text) => {
+            return slugForReviewLink ? (
+              <Link
+                title={i18n.moment(review.created).format('lll')}
+                key={review.id}
+                to={reviewListURL({
+                  addonSlug: String(slugForReviewLink),
+                  id: review.id,
+                  src: location.query.src,
+                })}
+              >
+                {text}
+              </Link>
+            ) : (
+              text
+            );
+          },
+        ],
+      ];
+
+      const getReplacements = () => {
+        if (hasAdminPermission) {
+          arrReplacements.push([
+            'linkUserProfileStart',
+            'linkUserProfileEnd',
             (text) => {
-              return slugForReviewLink ? (
+              return !noAuthor && hasAdminPermission ? (
```

Comment:
So I believe you should only push to `replacements` when both conditions are satisfied.